### PR TITLE
Fh 3435 add new app

### PR DESCRIPTION
**Pull request by wei-lee**
This PR:

1. introduces a new env var `SYNC_MONGODB_URL`. If this value is set, the sync server will use it as the mongodb connection string. This will allow other apps to connect to the mongodb of an existing app.
2. Introduce concurrencies for workers. Multiple workers of the same type can be created inside an app.
3. Disable the default backoff behaviour of the sync worker. This is to ensure the sync frequency will work. Otherwise the backoff will make the sync frequency un-predicated.

### CODE REVIEW

**Comment by wei-lee** [comment]
ping @david-martin @aidenkeating for review

File: lib/api.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ function FHapi(cfg) {
       return;
     }
     var poolSize = parseInt(process.env.SYNC_MONGODB_POOLSIZE) || 50;
+    console.log('sync.connect with mongodb = %s and redis = %s', connectionString, redisUrl);
```
Comment (david-martin) [inline lib/api.js]: @wei-lee should probably use 'debug' here

File: package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "fh-mbaas-api",
+  "name": "fh-mbaas-api-metrics",
```
Comment (david-martin) [inline package.json]: did you mean to make this change?

**Comment by david-martin** [comment]
@wei-lee These changes look reasonable.
A couple comments above.
Also, is there a way we can test these changes (unit and/or integration)?

**Comment by wei-lee** [comment]
@david-martin the comments you added are for an old commit I think. They are not in the PR anymore.

The existing integration tests already cover the changes around start/stop.

**Comment by wei-lee** [comment]
@david-martin ok to merge?

File: lib/sync/sync-server.js [hunk 3]
```diff
@@ -356,6 +356,8 @@ module.exports = {
     listCollisions: listCollisions,
     removeCollision: removeCollision,
     callHandler: callHandler,
-    getStats: getStats
+    getStats: getStats,
+    setRecordHashFn: hashProviderModule.setRecordHashFn,
```
Comment (david-martin) [inline lib/sync/sync-server.js]: are these 2 functions new to the API?

Comment (david-martin) [inline lib/sync/sync-server.js]: Also, this doesn't look related to the goal of the PR. 
We can include it if you want, but would be good to get more context

**Comment by david-martin** [comment]
@wei-lee  Merging now after conversation on IRC

```
<irldavem> weili_, re: https://github.com/feedhenry/fh-mbaas-api/pull/113, i don't mind if you include the hashing funcitons
<irldavem> but are the names of them correct?
<irldavem> i thought they were in the existing sync server too?
<irldavem> from Evans or Philips pr
<weili_> irldavem yes, but the existing one the name is `globalHandleHash`
<weili_> irldavem which is a bit confusing
<weili_> irldavem given that they are not yet documented
<weili_> irldavem I think we can take the opportunity to rename it
<irldavem> weili_, so document that they are added in 7.0.0?
<weili_> irldavem yes
```
